### ISSUE TITLE
Disallow Reserved Node Labels CHunk & Document in Graph Generation 

### ISSUE THREAD

**Issue by zach-blumenfeld** [neo4j-labs/llm-graph-builder#436]
The Graph Generator will infer "Document" labeled nodes when generating graphs. However, this conflicts with concept of the source Document node.  Is there any way to force the generator to not use already reserved node labels - specifically "Document" and "Chunk"? 

For example in the below only three documents where loaded into llm-grph-builder, but multiple other "Document" nodes were created
![image](https://github.com/neo4j-labs/llm-graph-builder/assets/89669746/c92aa8a7-7da3-4968-93bd-3ecc21cd00a1)

**Comment by jexp** [comment]
Yeah I know this is frustrating, we might need to use special `__Document__`, and `__Chunk__` labels in the future.
